stix_shifter_modules/cybereason/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,471 @@
+        return [value]
+
+    @staticmethod
+    def _parse_time_range(qualifier, time_range):
+        """
+        Converts qualifier timestamp to epoch
+        return: list of converted epoch values
+        """
+        try:
+            compile_timestamp_regex = re.compile(START_STOP_PATTERN)
+            converted_timestamp = []
+            if qualifier and compile_timestamp_regex.search(qualifier):
+                time_range_iterator = compile_timestamp_regex.finditer(qualifier)
+                time_range_list = [each.group() for each in time_range_iterator]
+            else:
+                start_time = STOP_TIME - timedelta(minutes=time_range)
+                converted_start_time = start_time.strftime('%Y-%m-%dT%H:%M:%S.%f')[:-3] + 'Z'
+                # limit 3 digit value for millisecond
+                converted_stop_time = STOP_TIME.strftime('%Y-%m-%dT%H:%M:%S.%f')[:-3] + 'Z'
+                time_range_list = [converted_start_time, converted_stop_time]
+            for timestamp in time_range_list:
+                converted_time = QueryStringPatternTranslator._format_datetime(timestamp)
+                converted_timestamp.append(converted_time)
+            return converted_timestamp
+        except (KeyError, IndexError, TypeError) as e:
+            raise e
+
+    def __eval_comparison_value(self, expression, mapped_fields_array):
+        """
+        Function for parsing comparison expression value
+        :param expression: expression object, mapped_fields_array
+        :return: formatted expression value
+        """
+        value = None
+        if expression.comparator == ComparisonComparators.In:
+
+            value = self._format_set(expression.value, mapped_fields_array)
+
+        elif expression.comparator in [ComparisonComparators.GreaterThan, ComparisonComparators.GreaterThanOrEqual,
+                                       ComparisonComparators.LessThan,
+                                       ComparisonComparators.LessThanOrEqual, ComparisonComparators.Equal,
+                                       ComparisonComparators.NotEqual]:
+            value = self._format_equality(expression.value, mapped_fields_array)
+        elif expression.comparator in [ComparisonComparators.Like, ComparisonComparators.Matches]:
+            value = self._format_like(expression.value, mapped_fields_array)
+        return value
+
+    @staticmethod
+    def _formatted_query(add_query_path, add_custom_fields, options):
+        """
+        Cybereason native data source format
+        """
+        query = {
+            "queryPath": add_query_path,
+            "queryLimits": {
+                "groupingFeature": {
+                    "elementInstanceType": add_query_path[0]["requestedType"],
+                    "featureName": "elementDisplayName"
+                }
+            },
+            "perFeatureLimit": 1,
+            "totalResultLimit": options["result_limit"] - 1,
```

Comment:
why are you subtracting 1 from the limit?

Reply:
The Cybereason datasource is returning extra one record for each query when using the options , perFeatureLimit = 1, perGroupLimit =1 and totalResultLimit = 10,000.
Hence options["result_limit"] -1  is used to fetch exactly 10,000 records.